utility/servicer/module.go

```diff
@@ -28,30 +45,325 @@ func CreateServicer(bus modules.Bus, options ...modules.ModuleOption) (modules.S
+func (s servicer) validateServicer(meta *coreTypes.RelayMeta, session *coreTypes.Session) error {
+	if meta.ServicerPublicKey != s.config.PublicKey {
+		return fmt.Errorf("relay servicer key %s does not match this servicer instance %s", meta.ServicerPublicKey, s.config.PublicKey)
+	}
+
+	var found bool
+	for _, servicer := range session.Servicers {
+		if servicer != nil && servicer.PublicKey == meta.ServicerPublicKey {
+			found = true
+			break
+		}
+	}
+
+	if !found {
+		return fmt.Errorf("relay servicer key %s not found in session %d with %d servicers", meta.ServicerPublicKey, session.SessionNumber, len(session.Servicers))
+	}
+
+	// TODO: implement isServicerMaxedOut
+	return nil
+}
+
+// admitRelay decides whether the relay should be served
+func (s servicer) admitRelay(relay *coreTypes.Relay) error {
+	// TODO: utility module should initialize the servicer (if this module instance is a servicer)
+	const errPrefix = "Error admitting relay"
+
+	if relay == nil {
+		return fmt.Errorf("%s: relay is nil", errPrefix)
+	}
+
+	height := s.GetBus().GetConsensusModule().CurrentHeight()
+	if err := s.validateRelayMeta(relay.Meta, int64(height)); err != nil {
+		return fmt.Errorf("%w: %s", errValidateRelayMeta, err.Error())
+	}
+
+	// TODO: update the CLI to include ApplicationAddress(or Application Public Key) in the RelayMeta
+	session, err := s.GetBus().GetUtilityModule().GetSession(relay.Meta.ApplicationAddress, int64(height), relay.Meta.RelayChain.Id, relay.Meta.GeoZone.Id)
+	if err != nil {
+		return fmt.Errorf("%s: failed to get a session for height %d for relay meta %s: %w", errPrefix, height, relay.Meta, err)
+	}
+
+	// TODO: (REFACTOR) use a loop to run all validators: would also remove the need for passing the session around
+	if err := validateRelayBlockHeight(relay.Meta, session); err != nil {
+		return fmt.Errorf("%w: %s", errValidateBlockHeight, err.Error())
+	}
+
+	if err := s.validateApplication(relay.Meta, session); err != nil {
+		return fmt.Errorf("%s: relay failed application validation: %w", errPrefix, err)
+	}
+
+	if err := s.validateServicer(relay.Meta, session); err != nil {
+		return fmt.Errorf("%s: relay failed servicer instance validation: %w", errPrefix, err)
+	}
+
+	return nil
+}
+
+// IMPROVE: Add session height tolerance to account for session rollovers
+func validateRelayBlockHeight(relayMeta *coreTypes.RelayMeta, session *coreTypes.Session) error {
+	sessionStartingBlock := session.SessionNumber * session.NumSessionBlocks
+	sessionLastBlock := sessionStartingBlock + session.SessionHeight
+
+	if relayMeta.BlockHeight >= sessionStartingBlock && relayMeta.BlockHeight <= sessionLastBlock {
+		return nil
+	}
+
+	return fmt.Errorf("relay block height %d not within session ID %s starting block %d and last block %d",
+		relayMeta.BlockHeight,
+		session.Id,
+		sessionStartingBlock,
+		sessionLastBlock)
+}
+
+// TECHDEBT: These structures were copied as placeholders from v0 and need to be updated to reflect changes in v1
```

Comment:
Can you move these into `service/relay.go` and make the TECHDEBT more explicit. We are likely going to change all of these but I want to keep deleting them outside the scope of this PR. 

 cc @adshmh for context